clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -264,8 +269,11 @@ public boolean hasValidVersion() {
         return oldestVersion() <= latestVersion();
     }
 
-    public Optional<ApiVersionsResponseData.ApiVersion> toApiVersion(boolean enableUnstableLastVersion) {
-        short oldestVersion = oldestVersion();
+    public Optional<ApiVersionsResponseData.ApiVersion> toApiVersion(boolean enableUnstableLastVersion,
+                                                                     Optional<ApiMessageType.ListenerType> listenerType) {
```

Comment:
Since listenerType is optional, could we add a comment when the caller is expected to pass in the right listenerType?

Reply:
Fair point, I updated this to have two methods so it's clear which one to be used when. One of them always take the API listener and the other one doesn't. There is only one non test usage of the latter currently.